packages/flutter_tools/lib/src/android/gradle_errors.dart

```diff
@@ -503,25 +503,30 @@ final GradleHandledError minCompileSdkVersionHandler = GradleHandledError(
   eventLabel: 'min-compile-sdk-version',
 );
 
+final RegExp _agpJavaError = RegExp(r'Android Gradle plugin requires Java\s+\d+\s+to run');
+
 @visibleForTesting
-final GradleHandledError jvm11RequiredHandler = GradleHandledError(
+final GradleHandledError incompatibleJavaAndAgpVersionsHandler= GradleHandledError(
```

Comment:
Consider linking to the code that handles this in AGP. 
https://cs.android.com/android-studio/platform/tools/base/+/mirror-goog-studio-main:build-system/gradle-core/src/main/java/com/android/build/gradle/internal/plugins/AndroidPluginBaseServices.kt?q=%22Android%20Gradle%20plugin%20requires%20Java%22

Reply:
done